### DISCUSSION TITLE
ARM64 thunks and relocation addends

### DISCUSSION THREAD

**Discussion by rbartlensky** [rui314/mold#667] [Q&A] [answered]
I was looking over #463 and while reading the code for adding thunks, I noticed that [`A`](https://github.com/rui314/mold/blob/8a3b291ee75c164636433d0b694e08d3a21a1338/elf/arch-arm64.cc#L150) is computed very differently when applying relocations, but when calculating whether a symbol is reachable by a branch instruction it is defined as [`A`](https://github.com/rui314/mold/blob/8a3b291ee75c164636433d0b694e08d3a21a1338/elf/arch-arm64.cc#L497). Is there any difference between the two, or is it just a small bug in the latter case?

**Comment by rui314** [comment, marked as answer]
I think it's not a bug. When applying a relocation, we don't know whether a relocation refers a section fragment or a symbol. A section fragment is a pieced of data in a mergeable section. On the other hand, when we are creating range extension thunks, we know we are handling only executable code, and we don't need to handle section pieces. So the simple `r.r_addend` suffices.

**Reply by rbartlensky** [reply]
I see, that makes a lot of sense -- thank you!
